File: Kotlin/신고 결과 받기.kt
fun solution(id_list: Array<String>, report: Array<String>, k: Int): IntArray {
    val map = mutableMapOf<String, Int>()
    val reportList = mutableMapOf<String, Int>()
    id_list.forEach {
        map[it] = 0
        reportList[it] = 0
    }
    report.distinct().forEach {
        map[it.split(" ")[1]] = map.getValue(it.split(" ")[1]) + 1
    }
    report.distinct().filter{ map.getValue(it.split(" ")[1]) >= k }.forEach{reportList[it.split(" ")[0]] = reportList.getValue(it.split(" ")[0]) + 1}
    return reportList.values.toIntArray()
}

File: Python/신고_결과_받기.py
def solution(id_list, report, k):
    answer = []
    report = list(set(report))
    num_user = len(id_list)

    reported = dict(zip(id_list, [0] * num_user))

    banned = list()
    for report_i in report:
        reported_user = report_i.split()[1]
        reported[reported_user] += 1

        if reported[reported_user] == k:
            banned.append(reported_user)

    ban_notice = dict(zip(id_list, [0] * num_user))
    for report_i in report:
        if report_i.split()[1] in banned:
            ban_notice[report_i.split()[0]] += 1

    for user in id_list:
        answer.append(ban_notice[user])

    return answer